# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/public/ginac/hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./ginsh/ginsh_parser.yy.orig	2005-02-22 13:22:31.395437640 +0100
+++ ./ginsh/ginsh_parser.yy	2005-02-22 13:23:02.008783704 +0100
@@ -797,7 +797,7 @@
 		// For shell commands, revert back to filename completion
 		rl_completion_append_character = orig_completion_append_character;
 		rl_basic_word_break_characters = orig_basic_word_break_characters;
-		rl_completer_word_break_characters = rl_basic_word_break_characters;
+		rl_completer_word_break_characters = (char*)rl_basic_word_break_characters;
 #if (GINAC_RL_VERSION_MAJOR < 4) || (GINAC_RL_VERSION_MAJOR == 4 && GINAC_RL_VERSION_MINOR < 2)
 		return completion_matches(const_cast<char *>(text), (CPFunction *)filename_completion_function);
 #else
@@ -807,7 +807,7 @@
 		// Otherwise, complete function names
 		rl_completion_append_character = '(';
 		rl_basic_word_break_characters = " \t\n\"#$%&'()*+,-./:;<=>?@[\\]^`{|}~";
-		rl_completer_word_break_characters = rl_basic_word_break_characters;
+		rl_completer_word_break_characters = (char*)rl_basic_word_break_characters;
 #if (GINAC_RL_VERSION_MAJOR < 4) || (GINAC_RL_VERSION_MAJOR == 4 && GINAC_RL_VERSION_MINOR < 2)
 		return completion_matches(const_cast<char *>(text), (CPFunction *)fcn_generator);
 #else
